src/server.rs

```diff
@@ -402,3 +418,47 @@ pub extern "C" fn rustls_server_session_write_tls(
         rustls_result::Ok
     }
 }
+
+/// Copy the SNI hostname to `buf` wich can hold up  to `count` bytes,
+/// and the length of that hostname in `out_n`. The string is stored in UTF-8
+/// with no terminating NUL byte.
+/// Returns RUSTLS_RESULT_INSUFFICIENT_SIZE if the SNI hostname is longer than `count`.
```

Comment:
Docs should also say "Returns Ok with *out_n == 0 if there is no SNI hostname available on this session because it hasn't been processed yet, or because the client did not send SNI."